pkg/query/querier.go

```diff
@@ -208,7 +208,19 @@ func (q *querier) Select(_ bool, hints *storage.SelectHints, ms ...*labels.Match
 	// The querier has a context but it gets canceled, as soon as query evaluation is completed, by the engine.
 	// We want to prevent this from happening for the async storea API calls we make while preserving tracing context.
 	ctx := tracing.CopyTraceContext(context.Background(), q.ctx)
-	ctx, cancel := context.WithTimeout(ctx, q.selectTimeout)
+
+	// Check if per-request timeout is specified
+	perRequestCancelTime, ifSpecified := q.ctx.Deadline()
+
+	var perQueryTimeOutValue time.Duration
```

Comment:
```suggestion
	var perQueryTimeout time.Duration
```
We know it's value, right?

Reply:
placeholder to contain the timeout value (to apply) in both the possibilities (per-request timeout is / is not specified